testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/SecurityAuthCommandsTestCase.java

```diff
@@ -0,0 +1,876 @@
+            } finally {
+                try {
+                    ctx.handle("reload");
+                } finally {
+                    ctx.terminateSession();
+                }
+            }
+        }
+        if (e != null) {
+            throw e;
+        }
+    }
+
+    @Test
+    public void testOOBSASLManagement() throws Exception {
+        ctx.handle("security enable-sasl-management --no-reload");
+        Assert.assertEquals(ElytronUtil.OOTB_MANAGEMENT_SASL_FACTORY, getManagementInterfaceAuthFactory(ctx, null, true));
+        ctx.handle("security disable-sasl-management --no-reload");
+        Assert.assertNull(getManagementInterfaceAuthFactory(ctx, null, true));
+    }
+
+    @Test
+    public void testOOBSASLManagementItf() throws Exception {
+        ctx.handle("security enable-sasl-management --no-reload --management-interface=" + Util.HTTP_INTERFACE);
+        Assert.assertEquals(ElytronUtil.OOTB_MANAGEMENT_SASL_FACTORY, getManagementInterfaceAuthFactory(ctx, Util.HTTP_INTERFACE, true));
+        ctx.handle("security disable-sasl-management --no-reload --management-interface=" + Util.HTTP_INTERFACE);
+        Assert.assertNull(getManagementInterfaceAuthFactory(ctx, Util.HTTP_INTERFACE, true));
+    }
+
+    @Test
+    public void testOOBSASLManagementNative() throws Exception {
+        enableNative(ctx);
+        try {
+            ctx.handle("security enable-sasl-management --no-reload --management-interface=" + Util.NATIVE_INTERFACE);
+            Assert.assertEquals(ElytronUtil.OOTB_MANAGEMENT_SASL_FACTORY, getManagementInterfaceAuthFactory(ctx, Util.NATIVE_INTERFACE, true));
+            ctx.handle("security disable-sasl-management --no-reload --management-interface=" + Util.NATIVE_INTERFACE);
+            Assert.assertNull(getManagementInterfaceAuthFactory(ctx, Util.NATIVE_INTERFACE, true));
+        } finally {
+            disableNative(ctx);
+        }
+    }
+
+    @Test
+    public void testOOBSASLManagement2() throws Exception {
+        // New factory but reuse OOB ManagementRealm properties realm.
+        // side effect is to create a constant realm mapper for ManagementRealm.
+        ctx.handle("security enable-sasl-management --no-reload --mechanism=DIGEST-MD5 "
+                + "--user-properties-file=mgmt-users.properties --group-properties-file=mgmt-groups.properties  "
+                + "--relative-to=jboss.server.config.dir --exposed-realm=ManagementRealm --new-security-domain-name=" + TEST_DOMAIN
+                + " --new-auth-factory-name=" + TEST_SASL_FACTORY);
+        Assert.assertEquals(TEST_SASL_FACTORY, getManagementInterfaceAuthFactory(ctx, null, true));
+
+        // Check Realm.
+        Assert.assertEquals("ManagementRealm", getExposedRealmName(true, "DIGEST-MD5"));
+        Assert.assertEquals("ManagementRealm", getMechanismRealmMapper(true, "DIGEST-MD5"));
+        getNames(Util.CONSTANT_REALM_MAPPER).contains("ManagementRealm");
+
+        // Replace with file system realm.
```

Comment:
This is not clear to me. How that it update TEST_DOMAIN ?

Reply:
The first call to enable-sasl-management creates the TEST_DOMAIN and associates the SASL factory to http-interface.
The second call to enable-sasl-managament detects that a factory is already present and updates it as well as the security domain it is associated to.